src/utils/git.ts

```diff
@@ -0,0 +1,42 @@
+import simpleGit, { SimpleGit } from 'simple-git';
+
+import { getConfigFileDir } from '../config';
+import { ConfigurationError } from './errors';
+import { logger } from '../logger';
+
+export async function getDefaultBranch(
+  git: SimpleGit,
+  remoteName: string
+): Promise<string> {
+  // This part is courtesy of https://stackoverflow.com/a/62397081/90297
+  return stripRemoteName(
+    await git.revparse(['--abbrev-ref', `${remoteName}/HEAD`]),
+    remoteName
+  );
+}
+
+export function stripRemoteName(
+  branch: string | undefined,
+  remoteName: string
+): string {
+  const branchName = branch || '';
+  const remotePrefix = remoteName + '/';
+  if (branchName.startsWith(remotePrefix)) {
+    return branchName.slice(remotePrefix.length);
+  }
+  return branchName;
+}
+
+export async function getGitClient(): Promise<SimpleGit> {
```

Comment:
Possible future refactor: replace direct `simpleGit` usage throughout the codebase with calls to `getGitClient`.

```
$ ag simpleGit
src/targets/ghPages.ts
5:import simpleGit from 'simple-git';
154:    await simpleGit().clone(remote.getRemoteStringWithAuth(), directory);
155:    const git = simpleGit(directory);

src/targets/upm.ts
2:import simpleGit from 'simple-git';
119:        const git = simpleGit(directory);

src/targets/crates.ts
4:import simpleGit from 'simple-git';
295:    const git = simpleGit(directory);

src/targets/awsLambdaLayer.ts
5:import simpleGit from 'simple-git';
149:        const git = simpleGit(directory);

src/targets/registry.ts
4:import simpleGit, { SimpleGit } from 'simple-git';
388:    const git = simpleGit(directory);

src/utils/git.ts
1:import simpleGit, { SimpleGit } from 'simple-git';
36:  const git = simpleGit(configFileDir);

src/config.ts
7:import simpleGit from 'simple-git';
250:    const git = simpleGit(configDir);
```

Reply:
Cannot really do it as most of the other usages involve a `.clone()` making the `isRepo` check obsolete.